diff -u ./onig-6.9.9/src/config.h ./config.h
--- ./onig-6.9.9/src/config.h	2023-10-14 19:58:22
+++ ./config.h	2023-10-14 20:17:11
@@ -11,7 +11,9 @@
 
 /* Define to 1 if you have <alloca.h> and it should be used (not on Ultrix).
    */
+#ifndef _WIN32
 #define HAVE_ALLOCA_H  1
+#endif
 
 /* Define to 1 if you have the <stdint.h> header file. */
 #define HAVE_STDINT_H  1
diff -u ./onig-6.9.9/src/oniguruma.h ./oniguruma.h
--- ./onig-6.9.9/src/oniguruma.h	2023-10-07 16:41:39
+++ ./oniguruma.h	2023-10-14 20:21:57
@@ -155,6 +157,7 @@
 typedef OnigEncodingType* OnigEncoding;
 
 ONIG_EXTERN OnigEncodingType OnigEncodingASCII;
+#ifdef ONIG_ALL_ENCODINGS
 ONIG_EXTERN OnigEncodingType OnigEncodingISO_8859_1;
 ONIG_EXTERN OnigEncodingType OnigEncodingISO_8859_2;
 ONIG_EXTERN OnigEncodingType OnigEncodingISO_8859_3;
@@ -170,7 +173,9 @@
 ONIG_EXTERN OnigEncodingType OnigEncodingISO_8859_14;
 ONIG_EXTERN OnigEncodingType OnigEncodingISO_8859_15;
 ONIG_EXTERN OnigEncodingType OnigEncodingISO_8859_16;
+#endif
 ONIG_EXTERN OnigEncodingType OnigEncodingUTF8;
+#ifdef ONIG_ALL_ENCODINGS
 ONIG_EXTERN OnigEncodingType OnigEncodingUTF16_BE;
 ONIG_EXTERN OnigEncodingType OnigEncodingUTF16_LE;
 ONIG_EXTERN OnigEncodingType OnigEncodingUTF32_BE;
@@ -185,8 +190,10 @@
 ONIG_EXTERN OnigEncodingType OnigEncodingCP1251;
 ONIG_EXTERN OnigEncodingType OnigEncodingBIG5;
 ONIG_EXTERN OnigEncodingType OnigEncodingGB18030;
+#endif
 
 #define ONIG_ENCODING_ASCII        (&OnigEncodingASCII)
+#ifdef ONIG_ALL_ENCODINGS
 #define ONIG_ENCODING_ISO_8859_1   (&OnigEncodingISO_8859_1)
 #define ONIG_ENCODING_ISO_8859_2   (&OnigEncodingISO_8859_2)
 #define ONIG_ENCODING_ISO_8859_3   (&OnigEncodingISO_8859_3)
@@ -202,7 +209,9 @@
 #define ONIG_ENCODING_ISO_8859_14  (&OnigEncodingISO_8859_14)
 #define ONIG_ENCODING_ISO_8859_15  (&OnigEncodingISO_8859_15)
 #define ONIG_ENCODING_ISO_8859_16  (&OnigEncodingISO_8859_16)
+#endif
 #define ONIG_ENCODING_UTF8         (&OnigEncodingUTF8)
+#ifdef ONIG_ALL_ENCODINGS
 #define ONIG_ENCODING_UTF16_BE     (&OnigEncodingUTF16_BE)
 #define ONIG_ENCODING_UTF16_LE     (&OnigEncodingUTF16_LE)
 #define ONIG_ENCODING_UTF32_BE     (&OnigEncodingUTF32_BE)
@@ -217,6 +226,7 @@
 #define ONIG_ENCODING_CP1251       (&OnigEncodingCP1251)
 #define ONIG_ENCODING_BIG5         (&OnigEncodingBIG5)
 #define ONIG_ENCODING_GB18030      (&OnigEncodingGB18030)
+#endif
 
 #define ONIG_ENCODING_UNDEF    ((OnigEncoding )0)
 
@@ -803,6 +813,8 @@
 ONIG_EXTERN
 int onig_initialize P_((OnigEncoding encodings[], int number_of_encodings));
 /* onig_init(): deprecated function. Use onig_initialize(). */
+ONIG_EXTERN
+int onig_is_initialized P_((void));
 ONIG_EXTERN
 int onig_init P_((void));
 ONIG_EXTERN
diff -u ./onig-6.9.9/src/regcomp.c ./regcomp.c
--- ./onig-6.9.9/src/regcomp.c	2023-10-10 16:30:30
+++ ./regcomp.c	2023-10-14 20:23:24
@@ -7778,6 +7778,12 @@
   return ONIG_NORMAL;
 }
 
+extern int
+onig_is_initialized()
+{
+  return onig_inited;
+}
+
 typedef struct EndCallListItem {
   struct EndCallListItem* next;
   void (*func)(void);
diff -u ./onig-6.9.9/src/regint.h ./regint.h
--- ./onig-6.9.9/src/regint.h	2023-05-20 08:08:34
+++ ./regint.h	2023-10-14 20:25:08
@@ -59,11 +59,13 @@
 
 /* config */
 /* spec. config */
+/*
 #define USE_REGSET
 #define USE_CALL
 #define USE_CALLOUT
-#define USE_BACKREF_WITH_LEVEL        /* \k<name+n>, \k<name-n> */
+#define USE_BACKREF_WITH_LEVEL         \k<name+n>, \k<name-n>
 #define USE_WHOLE_OPTIONS
+*/
 #define USE_RIGID_CHECK_CAPTURES_IN_EMPTY_REPEAT        /* /(?:()|())*\2/ */
 #define USE_NEWLINE_AT_END_OF_STRING_HAS_EMPTY_LINE     /* /\n$/ =~ "\n" */
 #define USE_WARNING_REDUNDANT_NESTED_REPEAT_OPERATOR
@@ -83,7 +85,9 @@
 #define USE_STRICT_POINTER_COMPARISON
 
 #define USE_WORD_BEGIN_END   /* "\<", "\>" */
+/*
 #define USE_CAPTURE_HISTORY
+*/
 #define USE_VARIABLE_META_CHARS
 #define USE_FIND_LONGEST_SEARCH_ALL_OF_RANGE
 /* #define USE_REPEAT_AND_EMPTY_CHECK_LOCAL_VAR */
